lib/modules/oec_group_comments/src/Plugin/Field/FieldFormatter/CommentGroupContentFormatter.php

```diff
@@ -0,0 +1,186 @@
+  }
+
+  /**
+   * Constructs a new CommentDefaultFormatter.
+   *
+   * @param string $plugin_id
+   *   The plugin_id for the formatter.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to which the formatter is associated.
+   * @param array $settings
+   *   The formatter settings.
+   * @param string $label
+   *   The formatter label display setting.
+   * @param string $view_mode
+   *   The view mode.
+   * @param array $third_party_settings
+   *   Third party settings.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityFormBuilderInterface $entity_form_builder
+   *   The entity form builder.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match object.
+   * @param \Drupal\Core\Entity\EntityDisplayRepositoryInterface $entity_display_repository
+   *   The entity display repository.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer interface.
+   * @param bool $is_xml_http_request
+   *   TRUE if the request is a XMLHttpRequest.
+   * @param \Drupal\oec_group_comments\GroupPermissionChecker $groupPermissionChecker
+   *   The group permission checker.
+   */
+  public function __construct(
+    $plugin_id,
+    $plugin_definition,
+    FieldDefinitionInterface $field_definition,
+    array $settings,
+    $label,
+    $view_mode,
+    array $third_party_settings,
+    AccountInterface $current_user,
+    EntityTypeManagerInterface $entity_type_manager,
+    EntityFormBuilderInterface $entity_form_builder,
+    RouteMatchInterface $route_match,
+    EntityDisplayRepositoryInterface $entity_display_repository,
+    RendererInterface $renderer,
+    $is_xml_http_request,
+    GroupPermissionChecker $groupPermissionChecker
+  ) {
+    parent::__construct(
+      $plugin_id,
+      $plugin_definition,
+      $field_definition,
+      $settings,
+      $label,
+      $view_mode,
+      $third_party_settings,
+      $current_user,
+      $entity_type_manager,
+      $entity_form_builder,
+      $route_match,
+      $entity_display_repository
+    );
+
+    $this->renderer = $renderer;
+    $this->isXmlHttpRequest = $is_xml_http_request;
+    $this->groupPermissionChecker = $groupPermissionChecker;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $output = parent::viewElements($items, $langcode);
+    $entity = $items->getEntity();
+
+    // Exclude entities without the set id.
+    if (!empty($entity->id())) {
+      $group_contents = GroupContent::loadByEntity($entity);
+    }
+
+    // Check if the content is posted in groups.
+    if (!empty($group_contents)) {
```

Comment:
What do you think about doing the inverse logic ? For example

```suggestion
    if (empty($group_contents)) {
        return $output;
    }
```

I found it more readable but it's a personal opinion I guess

Reply:
Agreed. Will update this.